c/tskit/tables.c

```diff
@@ -6780,40 +6818,75 @@ static int
 cmp_edge_cl(const void *a, const void *b) {
     const tsk_edge_t *ia = (const tsk_edge_t *) a;
     const tsk_edge_t *ib = (const tsk_edge_t *) b;
-    int ret = (ia->child > ib->child) - (ia->child < ib->child);
+    int ret = (ia->parent > ib->parent) - (ia->parent < ib->parent);
     if (ret == 0)  {
-        ret = (ia->left > ib->left) - (ia->left < ib->left);
+        ret = (ia->child > ib->child) - (ia->child < ib->child);
+        if (ret == 0)  {
+            ret = (ia->left > ib->left) - (ia->left < ib->left);
+        }
     }
     return ret;
 }
 
 /* Squash the edges in the specified array in place. The output edges will
  * be sorted by (child_id, left).
  */
+
 int TSK_WARN_UNUSED
-tsk_squash_edges(tsk_edge_t *edges, size_t num_edges, size_t *num_output_edges)
+tsk_squash_edges(tsk_edge_t *edges, size_t num_edges, tsk_size_t *num_output_edges)
 {
     int ret = 0;
     size_t j, k, l;
-    tsk_edge_t e;
+
+    if (num_edges == 0 || num_edges == 1){
+        *num_output_edges = (tsk_size_t) num_edges;
+        return ret;
+    }
 
     qsort(edges, num_edges, sizeof(tsk_edge_t), cmp_edge_cl);
     j = 0;
     l = 0;
     for (k = 1; k < num_edges; k++) {
-        assert(edges[k - 1].parent == edges[k].parent);
-        if (edges[k - 1].right != edges[k].left || edges[j].child != edges[k].child) {
-            e = edges[j];
-            e.right = edges[k - 1].right;
-            edges[l] = e;
+
+        if (edges[k-1].parent != edges[k].parent || edges[k-1].right != edges[k].left
```

Comment:
space around arithmetic operators please: ``edges[k - 1].parent`` --- there's a bunch more here and below.